### DISCUSSION TITLE
How to remove auto word spacing

### DISCUSSION THREAD

**Discussion by Jerrycjw** [posquit0/Awesome-CV#514] [Q&A]
Hi there,

I find that if I write some items in the description part of cventry, it will automatically remove the space between words and do a spacing itself, it looks good in printing, but if we want to parse this pdf file, the spacing is gone, every word is connected without space.

I'm wondering if you could tell me how to turn off this auto spacing function.
Thanks!

**Comment by posquit0** [comment]
Please upload working example to check your problem :)

**Comment by SpaceRangerWes** [comment]
@posquit0 I made an example for you to see. The steps I took were
1. Open through OverLeaf
2. Select and Download PDF
3. Open in Preview (OSX)
4. Select all, copy, and paste straight to a Google doc.

https://docs.google.com/document/d/1Q6Cogxyf4vbhrMkrbxl8cZCdy2QA30BJGMI34k-o9bY/edit?usp=sharing

**Comment by creatine-chaos** [comment]
I tried reproducing this issue but couldn't. I'm using Chrome to open the pdf file. I lose all of the formattings when I copy and paste but the spaces between words in the description section are there.

**Comment by brooksambrose** [comment]
fwiw this happens within bullet points only (I think) and happens when uploading the pdf to Indeed.com, and likely on all Applicant Tracking Systems (ATS). For instance, you can see the concatenated strings when uploading the [example](https://github.com/posquit0/Awesome-CV/blob/master/examples/resume.pdf) pdf on https://www.jobscan.co.

**Comment by EricPeterson99** [comment]
Any solution for this issue?

**Comment by NiallDillane** [comment]
This worked for me, from a similar issue:

https://github.com/posquit0/Awesome-CV/issues/275#issuecomment-475865923

**Comment by chelseanbr** [comment]
Hi all, I found a solution that worked for me, got it from ChatGPT with the following **Prompt:**
"How do I add pdf-parsable spaces between all words for bulleted items in the LaTeX Awesome-CV?"

**Edited Response** (Had to change "cvlist" from the original response to "cvitems"):
To add PDF-parsable spaces between all words for bulleted items in the Awesome-CV LaTeX template, you can modify the `cvitems ` environment. The `cvitems ` environment is used to create the bulleted lists in the Awesome-CV template. We'll customize this environment to include parsable spaces between words.

Here's how you can achieve this:

1. Open the `awesome-cv.cls` file in a text editor.

2. Search for the definition of the `cvitems` environment. It should look something like this:

```latex
\newenvironment{cvitems}{
  \begin{itemize}[leftmargin=2ex, nosep, noitemsep]
    \vspace{-2mm}
}{
  \end{itemize}
}
```

3. Modify the `cvitems` environment to include parsable spaces between words. We'll use the `\hspace{}` command to add space between each word.

```latex
\newenvironment{cvitems}{
  \begin{itemize}[leftmargin=2ex, nosep, noitemsep, label=]
    \setlength\itemsep{0pt}
    \setlength\parskip{0pt}
    \setlength\parsep{0pt}
    \renewcommand\labelitemi{\hspace{0.5em}\textbullet\hspace{0.5em}}
}{
  \end{itemize}
}
```

In this modification, we set the `label` option to be empty and redefine the `labelitemi` command to include `\hspace{}` before and after the bullet symbol, effectively adding spaces between each bullet point and word.

4. Save the `awesome-cv.cls` file after making the changes.

5. In your main `.tex` document, use the updated Awesome-CV class:

```latex
\documentclass[11pt, a4paper]{awesome-cv}

\begin{document}

% Your CV content here, use the cvitems environment as usual
\section{Education}
\begin{cvitems}
  \item First item in the list.
  \item Second item in the list.
  \item Third item in the list.
\end{cvitems}

\end{document}
```

With these modifications, the `cvitems` environment in Awesome-CV will now display the bulleted items with spaces between words in the PDF output, making it copy-pasteable and searchable. The spaces between words will be handled automatically, and you don't need to manually adjust the spacing for each item.

**Comment by goodsamannan** [comment]
Hello @chelseanbr. That worked perfectly!

An existing issue is with commas (,) the source code is yet to cover it. Have any ideas around resolving this?
